controller/node_controller_test.go

```diff
@@ -656,9 +739,22 @@ func (s *TestSuite) TestSyncNode(c *C) {
 			err = imIndexer.Add(rm)
 			c.Assert(err, IsNil)
 		}
+
 		// sync node status
 		for nodeName, node := range tc.nodes {
-			err := nc.syncNode(getKey(node, c))
+			if nc.controllerID == node.Name {
+				enqueueNodeForMonitor := func(key string) {
+					nc.queue.Add(key)
+				}
+				mon, err := monitor.NewFakeNodeMonitor(nc.logger, nc.ds, node, enqueueNodeForMonitor)
```

Comment:
You can init this fake monitor in `newTestNodeController`

Reply:
addressed!